Not all punycode domains are parsed correctly

Thanks for maintaining Scala-URI! I've been using it for [NsLookup.io](https://www.nslookup.io) for a while now :)

**Describe the bug**
This domain exists and can be visited in a webbrowser, but fails to parse correctly: [http://👸🏽.cf](http://👸🏽.cf). 

**To Reproduce**
```scala
object ReproducePunycodeBug extends App {
  import io.lemonlabs.uri.Url

  println(Url.parseOption("http://i❤️.ws").get.toStringPunycode) // This is fine
  println(Url.parseOption("http://👸🏽.cf").get.toStringPunycode) // Throws an exception
}
```
throws
```
http://xn--i-7iq.ws
Exception in thread "main" java.lang.ExceptionInInitializerError
	at onlinenslookup.analysis.ReproducePunycodeBug.main(ReproducePunycodeBug.scala)
Caused by: java.lang.IllegalArgumentException: java.text.ParseException: An unassigned code point was found in the input 👸🏽
	at java.base/java.net.IDN.toASCIIInternal(IDN.java:275)
	at java.base/java.net.IDN.toASCII(IDN.java:123)
	at java.base/java.net.IDN.toASCII(IDN.java:152)
	at io.lemonlabs.uri.inet.PunycodeSupport.toPunycode(PunycodeSupport.scala:7)
	at io.lemonlabs.uri.inet.PunycodeSupport.toPunycode$(PunycodeSupport.scala:5)
	at io.lemonlabs.uri.DomainName.toPunycode(Host.scala:114)
	at io.lemonlabs.uri.DomainName.toStringPunycode(Host.scala:180)
	at io.lemonlabs.uri.UrlWithAuthority.toStringPunycode$$anonfun$1(Uri.scala:857)
	at io.lemonlabs.uri.Authority.toString(Authority.scala:73)
	at io.lemonlabs.uri.AbsoluteUrl.toStringWithConfig(Uri.scala:1008)
	at io.lemonlabs.uri.UrlWithAuthority.toStringPunycode(Uri.scala:857)
	at io.lemonlabs.uri.UrlWithAuthority.toStringPunycode$(Uri.scala:721)
	at io.lemonlabs.uri.AbsoluteUrl.toStringPunycode(Uri.scala:966)
	at onlinenslookup.analysis.ReproducePunycodeBug$.<clinit>(ReproducePunycodeBug.scala:7)
	... 1 more
Caused by: java.text.ParseException: An unassigned code point was found in the input 👸🏽
	at java.base/jdk.internal.icu.text.StringPrep.map(StringPrep.java:302)
	at java.base/jdk.internal.icu.text.StringPrep.prepare(StringPrep.java:425)
	at java.base/java.net.IDN.toASCIIInternal(IDN.java:273)
	... 14 more
```

--- a/jvm/src/main/scala/io/lemonlabs/uri/inet/PunycodeSupport.scala
+++ b/jvm/src/main/scala/io/lemonlabs/uri/inet/PunycodeSupport.scala
@@ -4,5 +4,5 @@ import java.net.IDN
 
 trait PunycodeSupport {
   def toPunycode(host: String): String =
-    IDN.toASCII(host)
+    IDN.toASCII(host, IDN.ALLOW_UNASSIGNED)
 }

--- a/shared/src/test/scala/io/lemonlabs/uri/GithubIssuesTests.scala
+++ b/shared/src/test/scala/io/lemonlabs/uri/GithubIssuesTests.scala
@@ -113,4 +113,10 @@ class GithubIssuesTests extends AnyFlatSpec with Matchers with OptionValues {
       Success(RelativeUrl(AbsolutePath.fromParts("this[1]", "does", "not"), QueryString.empty, None))
     )
   }
+
+  "Github Issue #424" should "turn emojis into punycode" in {
+    Url.parseTry("http://👸🏽.cf").map(_.toStringPunycode) should equal(
+      Success("http://xn--on8hvh.cf")
+    )
+  }
 }

